src/components/Version/PreversionView.tsx

```diff
@@ -0,0 +1,201 @@
+import { AssignmentTypes } from '@/defaults/assignmentTypes'
+import { useRegistry } from '@/hooks/useRegistry'
+import { Button } from '@ttab/elephant-ui'
+import { CalendarIcon, MessageCircleMore, Text, Tag, SignalMedium } from '@ttab/elephant-ui/icons'
+import { AssigneeAvatars } from '../DataItem/AssigneeAvatars'
+import { TimeDisplay } from '../DataItem/TimeDisplay'
+import type { DocumentVersion } from '@ttab/elephant-api/repository'
+import type { EleBlock, EleDocument } from '@/shared/types'
+import type { AssignmentMeta } from '@/views/Assignments/types'
+import { PreversionViewInfo } from './PreversionViewInfo'
+
+export const PreversionView = ({
+  content,
+  previewVersion,
+  versionHistory
+}: {
+  content: EleDocument
+  previewVersion: bigint | undefined
+  versionHistory?: DocumentVersion[]
+}) => {
+  const { locale, timeZone } = useRegistry()
+
+  function getDescriptions(block: EleDocument | EleBlock) {
+    if (!block) {
+      return []
+    }
+    if ('meta' in block) {
+      return block?.meta?.['core/description']
+        ?.filter((desc: EleBlock) => desc?.data?.text?.length > 0)
+        ?.map((desc: EleBlock) => ({ text: desc.data.text, role: desc.role }))
+    }
+    return []
+  }
+
+  function getSlugline(block: EleBlock | AssignmentMeta): string {
+    if (block) {
+      return 'value' in block ? block?.value : ''
+    }
+    return ''
+  }
+
+  function formatDate(dateString: string) {
+    if (!dateString) {
+      return ''
+    }
+
+    return new Intl.DateTimeFormat(locale, {
+      weekday: 'short',
+      month: 'short',
+      day: 'numeric',
+      timeZone
+    }).format(new Date(dateString))
+  }
+
+  const assignmentType = (block: EleBlock) => block.meta['core/assignment-type']?.[0]?.value
+  const getAuthors = (block: EleBlock) => block.links['core/author']?.map((a: EleBlock) => a?.title)
+
+  const descriptions = getDescriptions(content)
+  const planningDate = content.meta['core/planning-item']?.[0]?.data
+  const eventDate = content.meta['core/event']?.[0]?.data
+  const slugline = getSlugline(content?.meta?.['tt/slugline']?.[0])
+  const newsvalue = content.meta['core/newsvalue']?.[0]?.value
+  const assignments = content.meta['core/assignment']?.map((a: EleBlock) => a)
+  const section = content.links['core/section']?.find((a: EleBlock) => a)?.title
+  const story = content.links['core/story']?.find((a: EleBlock) => a)?.title
```

Comment:
I dunno, but there's a lot of conditionally rendered components below, maybe we could do something like:
```suggestion
  const Story = () => {
    const value = content.links['core/story']?.find((a: EleBlock) => a)?.title
    return value ? <StyledButton>{value}</StyledButton> : null
  }
  
  ...
  <Story />
```
where applicable

Reply:
Refactored this file a bit, should look better now